crates/telio-wg/src/uapi.rs

```diff
@@ -559,9 +559,13 @@ fn parse_peer<R: Read>(
                     }
                 }
                 "preshared_key" => {
-                    peer.preshared_key = Some(val.parse().map_err(|e: KeyDecodeError| {
+                    let preshared: PresharedKey = val.parse().map_err(|e: KeyDecodeError| {
                         Error::ParsingError("preshared_key", e.to_string())
-                    })?);
+                    })?;
+
+                    if preshared.0 != [0; 32] {
```

Comment:
This is sort of unexpected case, I suggest adding a `trace` in here, might be very helpful when inspecting the logs